### PR TITLE
WT-9208 add cursor method to allow applications to detect checkpoint cursor mismatch

### PR DESCRIPTION
**Pull request by keithbostic**
Add undocumented cursor method to allow applications to detect checkpoint cursor mismatch.

### CODE REVIEW

File: src/cursor/cur_backup.c [hunk 1]
```diff
@@ -272,6 +272,7 @@ __wt_curbackup_open(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *other,
       __wt_cursor_notsup,                             /* largest_key */
       __wt_cursor_notsup,                             /* cache */
       __wt_cursor_reopen_notsup,                      /* reopen */
+      __wt_cursor_checkpoint_id,                      /* checkpoint ID */
```
Comment (quchenhao) [inline src/cursor/cur_backup.c:275]: What will other type of cursors got for this value? The latest checkpoint id? Then they will get different values from the same cursor if called multiple times?

Comment (keithbostic) [reply src/cursor/cur_backup.c:275]: > What will other type of cursors got for this value? The latest checkpoint id? Then they will get different values from the same cursor if called multiple times?

As documented, they'll get either 0 (if they're not a checkpoint cursor), or the checkpoint ID of the checkpoint they were opened on. Calling them repeatedly will return the same value, the value is set on open and doesn't change.

**Comment by quchenhao** [comment]
@keithbostic Is the ticket number correct? WT-9200 is about durable timestamp.

**Comment by keithbostic** [comment]
@quchenhao 
> Is the ticket number correct? [WT-9200](https://jira.mongodb.org/browse/WT-9200) is about durable timestamp.

You're correct; it should have been WT-9208, I've corrected the title. I've left the branch the same, I don't think the branch name matters. -- Thanks!

**Review by quchenhao — COMMENTED** [review]
Do we need to do anything to expose this to python api so we can test this?

**Comment by keithbostic** [comment]
> Do we need to do anything to expose this to python api so we can test this?

I didn't bother -- it's undocumented, with one consumer, test/format.

**Review by quchenhao — APPROVED** [review]
lgtm

**Pull request merged**